index.js

```diff
@@ -6,9 +6,18 @@ const { GitHub, context } = require('@actions/github')
 const main = async () => {
   const token = core.getInput('github-token')
   const number = core.getInput('number')
-
   const octokit = new GitHub(token)
 
+  // Check for existing approvals before approving a second time
+  const { data: reviews } = await octokit.pulls.listReviews({
+    ...context.repo,
+    pull_number: number
+  })
```

Comment:
I would recommend to use Graphql (see below), but if not, I'd set `per_page` to 100 which gives you the maximum amount of reviews per request, or even better, paginate the results, so you'll get all of them, even if there are more than 100

```suggestion
  const reviews = await octokit.paginate(octokit.pulls.listReviews, {
    ...context.repo,
    pull_number: number
    per_page: 100
  })
```

Reply:
Thanks for the detailed info. I had a feeling this was a tricky subject. 😬 

Would something like `viewerDidAuthor` be present on the REST API response? No, right?

Reply:
doesn't look like it https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-reviews-for-a-pull-request